### PR TITLE
[BrightcoveNew] Fix: find videos embedded in xilinx.com site

### PR DESCRIPTION
**Pull request by raron**
In xilinx.com sites there are no video tags, the video_id can be found in a div tag.
(the brightcove script is before the div (video-id) tag.)

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The xilinx.com site uses embedded brightcove videos, but there is no html video tag, so the original BrightcoveNewIE could not determine the video_id.
These sites uses a script tag (similar to what the original code search for), and after that there is a div tag, where the data-video-id (video_id) can be found.

Example sites:
https://www.xilinx.com/video/soc/how-to-use-the-zynq-7000-verification-ip-verify-debug-simulation.html
https://www.xilinx.com/video/hardware/model-composer-product-overview.html

### CODE REVIEW

**Comment by dstftw** [comment]
This is not a generic embed thus should go into separate extractor.
Core tests are broken.

**Comment by raron** [comment]
Should I use this branch, or create a new one (called xilinx) and delete this branch / pull request?

**Comment by dstftw** [comment]
Up to you.

**Comment by raron** [comment]
Done.

File: youtube_dl/extractor/xilinx.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            'ext': 'mp4',
+            'title': 'Model Composer Product Overview',
+            'description': 'md5:806e3831788848342777cdc3947c3d58',
+            'timestamp': 1513121997,
+            'upload_date': '20171212',
+            'uploader_id': '17209957001',
+            'categories': 'hardware',
+        },
+        'params': {
+            't': True,
```
Comment (dstftw) [inline youtube_dl/extractor/xilinx.py]: What's this?

File: youtube_dl/extractor/xilinx.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        'url': 'https://www.xilinx.com/video/hardware/model-composer-product-overview.html',
+        'info_dict': {
+            'id': '5678303886001',
+            'ext': 'mp4',
+            'title': 'Model Composer Product Overview',
+            'description': 'md5:806e3831788848342777cdc3947c3d58',
+            'timestamp': 1513121997,
+            'upload_date': '20171212',
+            'uploader_id': '17209957001',
+            'categories': 'hardware',
```
Comment (dstftw) [inline youtube_dl/extractor/xilinx.py]: Must be a list.

File: youtube_dl/extractor/xilinx.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
+
+    def _real_extract(self, url):
+
+        page_id = self._match_id(url)
+        category = re.match(self._VALID_URL, url).group('cat')
+        webpage = self._download_webpage(url, page_id)
+
+        urls = []
+        for account_id, player_id, embed in re.findall(
```
Comment (dstftw) [inline youtube_dl/extractor/xilinx.py]: Using `findall` and processing only first URL then is totally pointless.

File: youtube_dl/extractor/xilinx.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        if (len(urls) == 0):
+            self.report_warning("Couldn't get any video urls.")
+
+        if (len(urls) > 1):
+            self.report_warning("Got more than one video urls, using the first one.")
+
+        return {
+            '_type': 'url_transparent',
+            'url': urls[0],
```
Comment (dstftw) [inline youtube_dl/extractor/xilinx.py]: Breaks if no URLs extracted.

**Pull request closed without merging**